yaksh/models.py

```diff
@@ -720,8 +723,43 @@ class CourseStatus(models.Model):
     course = models.ForeignKey(Course)
     user = models.ForeignKey(User)
     grade = models.CharField(max_length=255, null=True, blank=True)
-    total_marks = models.FloatField(default=0.0)
+    percentage = models.FloatField(default=0.0)
 
+    def get_grade(self):
+        return self.grade
+
+    def set_grade(self):
+        if self.is_course_complete():
+            self.calculate_percentage()
+            if self.course.grading_system is None:
+                grading_system = GradingSystem.objects.get(name='default')
+            else:
+                grading_system = self.course.grading_system
+            grade = grading_system.get_grade(self.percentage)
+            self.grade = grade
+
+    def calculate_percentage(self):
+        if self.is_course_complete():
+            quizzes = self.course.get_quizzes()
+            total_weightage = 0
+            sum = 0
+            for quiz in quizzes:
+                total_weightage += quiz.weightage
+                marks = AnswerPaper.objects.get_user_best_of_attempts_marks(
+                        quiz, self.user.id, self.course.id)
+                out_of = quiz.questionpaper_set.first().total_marks
+                sum += (marks/out_of)*quiz.weightage
+            self.percentage = (sum/total_weightage)*100
+
+
+    def is_course_complete(self):
+        modules = self.course.get_learning_modules()
+        complete = False
+        for module in modules:
+            complete = module.get_status(self.user, self.course) == 'completed'
```

Comment:
Here I guess you have to check the condition if module's check_prerequisite is set to `True` otherwise it will return `False` even though all the other modules are completed.

Reply:
I did not understand clearly what this means!
Please explain in detail.
I am simply using the method ```get_status``` to check the status!
Where and how check_prerequisite comes into picture here?

Reply:
There should be a way by which we can say not check a particular module's completion status similar to check_prerequisite because some modules might just be introduction modules containing no units. In your case, that module's completion status is also checked and since it is not completed the course will show `in progress`.  So, for now, `check_prerequisite` serves this purpose. So I thought we can add a condition to check if module `check_prerequisite` is set to True. We can discuss it.

Reply:
I think then it should be in module ```get_status``` method.
What do you think?
Because this is simply using the get_status method to check if the module is completed or not.
```get_status``` should return  completed if module has no unit!
May be we can add this in a different PR!
Created issue #440

Reply:
I have commented on #440 just check if that is ok to implement